src/node/audio_buffer_source.rs

```diff
@@ -161,100 +152,94 @@ impl AudioBufferSourceNode {
-                // actual buffer number_of_channels when received on the render thread.
-                internal_buffer: Vec::<f32>::with_capacity(2),
+                // `internal_buffer` is used to compute the samples per channel at each frame. Note
+                // that the `vec` will always be resized to actual buffer number_of_channels when
+                // received on the render thread.
+                internal_buffer: Vec::<f32>::with_capacity(crate::MAX_CHANNELS),
             };
 
-            let mut node = Self {
+            let node = Self {
                 registration,
                 controller,
                 channel_config: channel_config.into(),
                 sender,
                 detune: d_param,
                 playback_rate: pr_param,
-                buffer: None,
-                source_started: false,
+                buffer: OnceCell::new(),
+                source_started: AtomicBool::new(false),
             };
 
             node.controller.set_loop(loop_);
             node.controller.set_loop_start(loop_start);
             node.controller.set_loop_end(loop_end);
 
             if let Some(buf) = buffer {
-                node.set_buffer(&buf);
+                node.set_buffer(buf);
             }
 
             (node, Box::new(renderer))
         })
     }
 
+    /// Current buffer value (nullable)
+    pub fn buffer(&self) -> Option<&AudioBuffer> {
+        self.buffer.get()
+    }
+
     /// Provide an [`AudioBuffer`] as the source of data to be played bask
     ///
     /// # Panics
     ///
-    /// Panics if a buffer has already been given to the source (though `new`
-    /// or through `set_buffer`)
-    pub fn set_buffer(&mut self, audio_buffer: &AudioBuffer) {
-        // - Let new buffer be the AudioBuffer or null value to be assigned to buffer.
-        // - If new buffer is not null and [[buffer set]] is true, throw an
-        // InvalidStateError and abort these steps.
-        if self.buffer.is_some() {
+    /// Panics if a buffer has already been given to the source (though `new` or through
+    /// `set_buffer`)
+    pub fn set_buffer(&self, audio_buffer: AudioBuffer) {
+        let clone = audio_buffer.clone();
+
+        if self.buffer.set(audio_buffer).is_err() {
             panic!("InvalidStateError - cannot assign buffer twice");
         }
-        // - If new buffer is not null, set [[buffer set]] to true.
-        // - Assign new buffer to the buffer attribute.
-        self.buffer = Some(audio_buffer.clone());
-        // - If start() has previously been called on this node, perform the
-        // operation acquire the content on buffer.
-        // see <https://github.com/orottier/web-audio-api-rs/issues/68> for
-        // more information on the "acquire the content" concept in this context.
-        if self.source_started {
-            self.send_buffer();
-        }
+
+        self.sender
```

Comment:
Notice I'm sending the buffer regardless of `self.source_started`. I think that is just fine, also regarding our discussion resulting in not needing acquire-the-content ops

Reply:
Yup, looks fine and more simple